lcls_naming_tool/web/app.py

```diff
@@ -0,0 +1,31 @@
+import sys
+import os
+from flask import Flask, render_template, request, url_for, flash, redirect
+from lcls_naming_tool import get_version, load_taxons, validate
+
+
+sys.path.append(os.path.abspath('../'))
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'your secret key'
+
+@app.route('/', methods=('GET', 'POST'))
+def index():
+
+    version = get_version()
+
+    if request.method == 'POST':
+        pv_name = str(request.form['pv_name']).upper()
+
+        flash(pv_name, 'pv_name')
+
+        load_taxons()
```

Comment:
This function loads the taxon dicts from file with each call.  This seems like wasteful I/O, is there a way to do this once for the lifetime of the flask app?

This raises questions of how we intend to update the taxon information, but I think having machinery around that (even if it's as simple as restarting the flask server) is worth the trouble.

Reply:
Moved load_taxons() outside of the index() function so it only gets called once per user session. 

Github runner would be a good solution to allow the site to auto-update whenever the repo changes.

Reply:
To be clear, until cyber security resolves the issue of local runners, our use of Github Actions is limited to cloud runners. These cloud runners cannot interact with our local hosts.   

The more practical solution currently is to have a cron job to update the files periodically.

Reply:
I think Alex mentioned that the user creating the name is likely to want to see it available immediately, but given that local runners aren't available, a cron job makes the most sense.